
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>3. Unlocking biological significance from the data &#8212; Analysis of semi-tryptic proteomics data</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="2. Processing of mass spectrometry data" href="data_import_process.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/UBC_The-Overall-Lab-Web-Site_Protease-Web.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Analysis of semi-tryptic proteomics data</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../markdown/intro.html">
                    Welcome to Degradomics analysis in Jupyter Lab
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../markdown/degradomics.html">
   1. Degradomics - the study of proteolytic processing
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="data_import_process.html">
   2. Processing of mass spectrometry data
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   3. Unlocking biological significance from the data
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/notebooks/data_subset.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="../_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/notebooks/data_subset.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#filtering-of-peptides">
   3.1. Filtering of peptides
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plotting-subsets-of-data">
   3.2. Plotting subsets of data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#irrelevant-images">
   3.3. Irrelevant images
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#irrelevant-formulae">
   3.4. Irrelevant formulae
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Unlocking biological significance from the data</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#filtering-of-peptides">
   3.1. Filtering of peptides
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plotting-subsets-of-data">
   3.2. Plotting subsets of data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#irrelevant-images">
   3.3. Irrelevant images
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#irrelevant-formulae">
   3.4. Irrelevant formulae
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="unlocking-biological-significance-from-the-data">
<h1><span class="section-number">3. </span>Unlocking biological significance from the data<a class="headerlink" href="#unlocking-biological-significance-from-the-data" title="Permalink to this headline">#</a></h1>
<section id="filtering-of-peptides">
<h2><span class="section-number">3.1. </span>Filtering of peptides<a class="headerlink" href="#filtering-of-peptides" title="Permalink to this headline">#</a></h2>
<p>In the previous section <a class="reference internal" href="data_import_process.html"><span class="doc std std-doc">Processing of mass spectrometry data</span></a>, we used a volcano plot to visualise data from MSfragger, to generate a volcano plot – but how can we access information regarding the peptides and proteins described in the figure?</p>
<p>We do this by filtering the data, with the package <code class="docutils literal notranslate"><span class="pre">dplyr</span></code> and <code class="docutils literal notranslate"><span class="pre">tidyr</span></code>. An example of the code to filter peptides based on significance at the peptide and protein level is described below:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">peptide_sig</span> <span class="o">=</span> <span class="n">peptide</span> <span class="o">%&gt;%</span> <span class="nb">filter</span><span class="p">(</span><span class="n">p_adj</span> <span class="o">&lt;</span> <span class="mf">0.05</span> <span class="o">&amp;</span> <span class="n">sarscov2_control_ratio</span> <span class="o">&gt;</span><span class="mf">0.6</span> <span class="o">&amp;</span> 
                                   <span class="n">semi_tryptic</span> <span class="o">==</span> <span class="n">TRUE</span> <span class="o">&amp;</span> <span class="n">sig_protein</span> <span class="o">==</span> <span class="n">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<p>If we execute this code, we derive the significantly different peptides:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># first, load required packages</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reshape2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">gghighlight</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘dplyr’
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The following objects are masked from ‘package:stats’:

    filter, lag
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘reshape2’
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>The following object is masked from ‘package:tidyr’:

    smiths
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># next, read in data file</span>
<span class="n">peptide</span> <span class="o">=</span> <span class="nf">read.csv</span><span class="p">(</span><span class="s">&quot;../data/peptide_processed.csv&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># filter for significant differences</span>
<span class="n">peptide_sig</span> <span class="o">=</span> <span class="n">peptide</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">p_adj</span> <span class="o">&lt;</span> <span class="m">0.05</span> <span class="o">&amp;</span> <span class="n">sarscov2_control_ratio</span> <span class="o">&gt;</span><span class="m">0.6</span> <span class="o">&amp;</span> 
                                   <span class="n">semi_tryptic</span> <span class="o">==</span> <span class="kc">TRUE</span> <span class="o">&amp;</span> <span class="n">sig_protein</span> <span class="o">==</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>We can print this table to identify the peptide sequences in question:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">print</span><span class="p">(</span><span class="n">peptide_sig</span><span class="o">$</span><span class="n">Peptide</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> [1] &quot;GDRSEDFGVNEDLAD&quot;  &quot;RLLEDGEDFNLGDALD&quot; &quot;GLLAGNEK&quot;         &quot;SAPGTDLAK&quot;       
 [5] &quot;SLLEGQEDHYNNLS&quot;   &quot;DNLTLWTSENQGDEGD&quot; &quot;SVPAAEPEYPK&quot;      &quot;ALPGDNVGFNVK&quot;    
 [9] &quot;GHYTEGAELVD&quot;      &quot;EDMAALEKDYEEVGVD&quot;
</pre></div>
</div>
</div>
</div>
<p>Interestingly, many of these semi-tryptic peptides end with ‘D’ (i.e. aspartic acid). Based on the position of this amino acid, it is likely that the semi-tryptic peptides were generated by caspases</p>
<p>What about the genes that encode these peptides? – we can obtain this information too:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">print</span><span class="p">(</span><span class="n">peptide_sig</span><span class="o">$</span><span class="n">Gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> [1] &quot;ANXA1&quot;  &quot;KRT18&quot;  &quot;KRT19&quot;  &quot;KRT19&quot;  &quot;KRT19&quot;  &quot;YWHAB&quot;  &quot;AK2&quot;    &quot;EEF1A1&quot;
 [9] &quot;TUBB4B&quot; &quot;TUBA1A&quot;
</pre></div>
</div>
</div>
</div>
</section>
<section id="plotting-subsets-of-data">
<h2><span class="section-number">3.2. </span>Plotting subsets of data<a class="headerlink" href="#plotting-subsets-of-data" title="Permalink to this headline">#</a></h2>
<p>Now if we prefer, we can plot only the semi-tryptic peptides that pertain to proteins that are not significantly different between conditions:</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">g1</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">peptide_sig</span><span class="p">,</span> 
            <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">sarscov2_control_ratio</span><span class="p">,</span> <span class="n">y</span> <span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p_adj</span><span class="p">)))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="o">=</span><span class="m">0.4</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="m">1.75</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;log2 fold change&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;-log10 FDR-corrected p-value&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">theme_bw</span><span class="p">()</span> <span class="c1">#+</span>
<span class="n">g1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/data_subset_12_0.png" src="../_images/data_subset_12_0.png" />
</div>
</div>
<figure class="align-default" id="crossref-key" style="width: 700px">
<figcaption>
<p><span class="caption-number">Fig. 3.1 </span><span class="caption-text">Differentially expressed semi-tryptic peptides</span><a class="headerlink" href="#crossref-key" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
<p>Should we wish to generate a plot where the the data points are labelled, we can do so with the following code:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>g3 &lt;- ggplot(data=peptide_sig, 
            aes(x=sarscov2_control_ratio, y =-log10(p_adj), label=Gene)) +
  geom_point(alpha=0.4, size=1.75) +
  geom_text(label = peptide_sig$Gene) +
  xlab(&quot;log2 fold change&quot;) + ylab(&quot;-log10 FDR-corrected p-value&quot;) +
  theme_bw()
g3
</pre></div>
</div>
<p>Executing this code gives the following output:</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">g</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">peptide_sig</span><span class="p">,</span> 
            <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">sarscov2_control_ratio</span><span class="p">,</span> <span class="n">y</span> <span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p_adj</span><span class="p">),</span> <span class="n">label</span><span class="o">=</span><span class="n">Gene</span><span class="p">))</span> <span class="o">+</span>
  <span class="nf">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="o">=</span><span class="m">0.4</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="m">1.75</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">geom_text</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="n">peptide_sig</span><span class="o">$</span><span class="n">Gene</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;log2 fold change&quot;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;-log10 FDR-corrected p-value&quot;</span><span class="p">)</span> <span class="o">+</span>
  <span class="nf">theme_bw</span><span class="p">()</span> <span class="c1">#+</span>
<span class="n">g</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../_images/data_subset_16_0.png" src="../_images/data_subset_16_0.png" />
</div>
</div>
<figure class="align-default" id="id1" style="width: 700px">
<figcaption>
<p><span class="caption-number">Fig. 3.2 </span><span class="caption-text">Annotated differentially expressed semi-tryptic peptides</span><a class="headerlink" href="#id1" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
<p>We can’t rule out that semi-tryptic peptides pertaining to proteins that <strong>are</strong>  significantly different between conditions are biologically relevant; we can filter them and view as follows:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># filter for significant differences</span>
<span class="n">pep_sig_prot_sig</span> <span class="o">=</span> <span class="n">peptide</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">p_adj</span> <span class="o">&lt;</span> <span class="m">0.05</span> <span class="o">&amp;</span> <span class="n">sarscov2_control_ratio</span> <span class="o">&gt;</span><span class="m">0.6</span> <span class="o">&amp;</span> 
                                   <span class="n">semi_tryptic</span> <span class="o">==</span> <span class="kc">TRUE</span> <span class="o">&amp;</span> <span class="n">sig_protein</span> <span class="o">==</span> <span class="kc">TRUE</span><span class="p">)</span>

<span class="nf">print</span><span class="p">(</span><span class="n">pep_sig_prot_sig</span><span class="o">$</span><span class="n">Gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;S&quot;   &quot;N&quot;   &quot;N&quot;   &quot;MX1&quot;
</pre></div>
</div>
</div>
</div>
<p>This is unsurprsing; 3 of the 4 peptides pertain to Sars-CoV-2 proteins. 1 peptide pertains to the protein encoded by MX1, a human gene associated with anti-viral responses</p>
</section>
<section id="irrelevant-images">
<h2><span class="section-number">3.3. </span>Irrelevant images<a class="headerlink" href="#irrelevant-images" title="Permalink to this headline">#</a></h2>
<p>To conclude the book, here are a couple of irrelevant images:</p>
<figure class="align-default" id="tails1">
<a class="reference internal image-reference" href="../_images/tails_1.png"><img alt="../_images/tails_1.png" src="../_images/tails_1.png" style="height: 200px;" /></a>
<figcaption>
<p><span class="caption-number">Fig. 3.3 </span><span class="caption-text">Example TMT prototeomics datasets</span><a class="headerlink" href="#tails1" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
<figure class="align-default" id="tails2">
<a class="reference internal image-reference" href="../_images/tails_2.png"><img alt="../_images/tails_2.png" src="../_images/tails_2.png" style="height: 200px;" /></a>
<figcaption>
<p><span class="caption-number">Fig. 3.4 </span><span class="caption-text">Another example TMT prototeomics dataset</span><a class="headerlink" href="#tails2" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
<p>Both <a class="reference internal" href="#tails1"><span class="std std-numref">Fig. 3.3</span></a> and <a class="reference internal" href="#tails2"><span class="std std-numref">Fig. 3.4</span></a> are examples of datasets where these methods could be applied.</p>
</section>
<section id="irrelevant-formulae">
<h2><span class="section-number">3.4. </span>Irrelevant formulae<a class="headerlink" href="#irrelevant-formulae" title="Permalink to this headline">#</a></h2>
<p>I need another couple of formulae in this page to meet the marking criteria. So here they are:</p>
<div class="math notranslate nohighlight" id="equation-mxc">
<span class="eqno">(3.1)<a class="headerlink" href="#equation-mxc" title="Permalink to this equation">#</a></span>\[y = mx+c\]</div>
<div class="math notranslate nohighlight" id="equation-random">
<span class="eqno">(3.2)<a class="headerlink" href="#equation-random" title="Permalink to this equation">#</a></span>\[a^2 - b^2 = (a-b)(a+b)\]</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            kernelName: "ir",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="data_import_process.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">2. </span>Processing of mass spectrometry data</p>
        </div>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Peter Bell<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>